src/main/java/com/denizenscript/depenizen/bukkit/commands/noteblockapi/NBSCommand.java

```diff
@@ -115,12 +131,28 @@ public static void autoExecute(ScriptEntry scriptEntry,
                 }
                 sp.setAutoDestroy(true);
                 sp.setPlaying(true);
+                if (scriptEntry.shouldWaitFor()) {
+                    waitingForSongs.put(sp, scriptEntry);
+                }
+                else {
+                   scriptEntry.setFinished(true);
```

Comment:
(can't comment directly on relevant spots, but) setFinished needs to be called in all cases, including any/all errors, otherwise it'll freezelock random queues on the server any time a mistake is made